server/api/annotation.py

```diff
@@ -39,15 +39,20 @@ def __init__(self):
         self.resourceName = 'annotation'
 
         self.route('GET', (), self.find)
-        self.route('GET', (':id',), self.getAnnotation)
-        self.route('GET', (':id', 'render'), self.renderAnnotation)
-        self.route('PUT', (':id',), self.submitAnnotation)
+        self.route('GET', (':annotationId',), self.getAnnotation)
+        self.route('GET', (':annotationId', ':featureId'), self.getAnnotationFeature)
+        self.route('GET', (':annotationId', ':featureId', 'mask'), self.getAnnotationFeatureMask)
+        self.route('GET', (':annotationId', ':featureId', 'render'),
+                   self.getAnnotationFeatureRendered)
+        self.route('PUT', (':annotationId',), self.submitAnnotation)
 
     @describeRoute(
         Description('Return a list of annotations.')
         .param('studyId', 'The ID of the study to filter by.', paramType='query', required=True)
         .param('userId', 'The ID of the user to filter by.', paramType='query', required=False)
         .param('imageId', 'The ID of the image to filter by.', paramType='query', required=False)
+        .param('state', 'Filter annotations to those at a given state', paramType='query',
```

Comment:
Nit: The other parameter descriptions end in a period, so this one should too.

Reply:
Fixing.